subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -254,6 +256,37 @@ public void execute(String openPullRequestBranch) {
             }
         });
 
+        TaskMaker.task(project, MERGE_PULL_REQUEST, MergePullRequestTask.class, new Action<MergePullRequestTask>() {
+            @Override
+            public void execute(final MergePullRequestTask task) {
+                task.setDescription("Merge pull request when all checks will be passed");
+                task.mustRunAfter(CREATE_PULL_REQUEST);
```

Comment:
@wwilk I'm not sure. What if check will fail for some reason? I assume there is possibility to fix something without opening new pull request, just over old one. 

Maybe we should base it on testing if pull request is open and run this method only if PullRequest is really open? We should 

    task.onlyIf(isPullRequestOpen());
    task.onlyIf(wasBuildFileUpdatedSpec(replaceVersionTask));

where isPullRequestOpen contains logic about PullRequest accessibility?

Reply:
You mean when check fails and it's not auto-merged? I guess it should happen much less often than the check succeeding so in this case we can create another PR. But anyway maybe it would be better to base .onlyIf on the existence of pull request to be merged. So if either FindOpenPullRequest returns one, or CreatePullRequest creates one, we merge. Otherwise we don't.